src/pages/Home/Home.tsx

```diff
@@ -21,91 +20,78 @@ export default function Home() {
   const [activeCategory, setActiveCategory] = useState<categoryPosition>({
     category: "",
     offsetY: 0,
-  }); //dependent on scroll position and categories
+  });
   const [categories, setCategories] = useState<categoryPosition[]>([]);
   const [scrollPosition, setScrollPosition] = useState(0);
+  const contentDivRef = useRef<HTMLDivElement>(null);
 
-  // add and remove scrollListener on Home mount and dismount. On scroll event, handleScroll is called, which updates scrollPosition.
+  const HEADER_HEIGHT = 119;
+
+  // onMount, add scrollListeners, Initialise categories, y-position, and default activeCategory.
   useEffect(() => {
     function scrollHandler() {
       setScrollPosition(window.scrollY);
     }
     window.addEventListener("scroll", scrollHandler);
-    console.log("Mounted");
+
+    //bugFix. I'm not sure why but if I try to get y-position immediately, the value is incorrect. Have to wait around 70ms, then it will work.
+    setTimeout(() => {
```

Comment:
Update on bug:

Bug occurs when I try to read y-position immediately after component mounts. By setTimeout, then reading y-position, the value becomes correct.
Maybe it's because getting the img in the content takes some time, so the dom is not updated immediately.

I don't know if this workaround is 'good practice'. It seems like a random setTimeout here.

Reply:
Yeah i think this doesn't look quite right.
Can you try this:
1. In Home.tsx get a list of the categories 
2. In the HTML component of each category, add an id with the category name to the section
3. In useEffect in Home.tsx, get the position of each element each time the user scrolls
4. With the positions of each section, calculate what's the top section that's available in the viewport

Another way
1. Use intersection Observer https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API to find out if the element is in viewport
2. If the element is in viewport, update the redux state, otherwise, remove it. So the redux slice should have a list of categories like [{categoryName: "Starter", inViewport: false}, {}]. So at the start, you populate the category names in the list. Then create a dispatch function that updates the list when the state of whether a category is in the viewport changes
4. Redux reducer should have a list of all categories that are in the dom. Sort them by the order of the category and then update the highlighted category accordingly
Don't get the category names from the DOM elements like firstChild.textContent. The names should come from the same place throughout the site, which is the DB.